File: PYTActividad2PG3/Clases/Automovil.py
from Clases.Vehiculo import *

class Automovil(Vehiculo):
    def __init__(self, FechaDEfabricacion, VIN_chasis, VINMotor, marca, modelo, precio):
        super().__init__(FechaDEfabricacion, VIN_chasis, VINMotor)
        self.marca = marca
        self.modelo = modelo
        self.precio = precio
    
    def obtener_marca(self):
        return self.marca

    def obtener_modelo(self):
        return self.modelo

    def obtener_precio(self):
        return self.precio

    def imprimir(self):
        print(f"Fecha de Fabricación: {self.obtener_FechaDEfabricacion()}")
        print(f"VIN del Chasis: {self.obtener_VIN_chasis()}")
        print(f"VIN del Motor: {self.obtener_VINMotor()}")
        print(f"Marca: {self.obtener_marca()}")
        print(f"Modelo: {self.obtener_modelo()}")
        print(f"Precio: ${self.obtener_precio():,.2f}")

File: HrnciaActividad2PG3/src/Clases/Automovil.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Clases;

/**
 *
 * @author luisd
 */
public class Automovil extends Vehiculo {

    private String marca;
    private String modelo;
    private double precio;

    public Automovil(String FechaDEfabricacion, String VIN_chasis, String VINMotor,String marca, String modelo, double precio) {
        super(FechaDEfabricacion, VIN_chasis, VINMotor);
        this.marca = marca;
        this.modelo = modelo;
        this.precio = precio;

    }

    public void mostrarDatos(){
        System.out.println("Fecha de fabricacion: " + getFechaDEfabricacion());
        System.out.println("Numero de chasis: " + getVIN_chasis());
        System.out.println("Numero de motor: " + getVINMotor());
        System.out.println("Marca : " + getMarca());
        System.out.println("Modelo: " + getModelo());
        System.out.println("Precio: " + getPrecio());
        
    }

    /**
     * @return the marca
     */
    public String getMarca() {
        return marca;
    }

    /**
     * @param marca the marca to set
     */
    public void setMarca(String marca) {
        this.marca = marca;
    }

    /**
     * @return the modelo
     */
    public String getModelo() {
        return modelo;
    }

    /**
     * @param modelo the modelo to set
     */
    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    /**
     * @return the precio
     */
    public double getPrecio() {
        return precio;
    }

    /**
     * @param precio the precio to set
     */
    public void setPrecio(double precio) {
        this.precio = precio;
    }



}
